procrastinate/job_context.py

```diff
@@ -64,6 +64,8 @@ class JobContext:
     additional_context: dict = attr.ib(factory=dict)
     task_result: Any = None
 
+    jobs_to_abort: set[int]
```

Comment:
If we ever use subprocesses for sync tasks, we should also consider the most straightforward way to provide this information to the subprocess. Or we can decide when the time comes.

Reply:
(That mainly highlights the importance of making this a "private" attribute (leading `_`). That said, I'm in favor of not trying to guess what we'll end up doing for subprocesses _yet_ :D

Reply:
Me, too 😞. At least not for a v3 release.

Reply:
> I wonder if we should just remove frozen=True because, well, the context is effectively not immutable, rather than cheating. I think I'd prefer that (even if I love immutable objects)
Or instead of providing a jobs_to_abort set, we'd give the context a should_abort callable...
Any opinion ?

I went through a similar thought process. 
I also prefer dropping the `frozen` flag and turning it into a boolean

Reply:
Thinking more about it, I am concerned that dropping frozen opens the door to all kinds of issues if the end user code starts changing that object. I will give the Callable option a try.